core/src/services/onedrive/backend.rs

```diff
@@ -96,11 +103,13 @@ impl Access for OnedriveBackend {
     }
 
     async fn write(&self, path: &str, args: OpWrite) -> Result<(RpWrite, Self::Writer)> {
-        let path = build_rooted_abs_path(&self.core.root, path);
-
         Ok((
             RpWrite::default(),
-            oio::OneShotWriter::new(OneDriveWriter::new(self.core.clone(), args, path)),
+            oio::OneShotWriter::new(OneDriveWriter::new(
```

Comment:
Since we already support chunked writing, we could implement something like `MultipartUploadWrite` instead of `OneShotWrite`, which only allows a single write.

Reply:
This PR is large enough, it's fine to split in another PR.

Reply:
Ok. I will fix your other comments this week and ping you. I'm tied up at the moment.